lupa/tests/test.py

```diff
@@ -632,6 +632,91 @@ def test_table_from_table_iter_indirect(self):
         self.assertEqual(list(table2.keys()), [1, 2, 3])
         self.assertEqual(set(table2.values()), set([1, 2, "foo"]))
 
+    def test_table_from_nested_dict(self):
+        data = {"a": {"a": "foo"}, "b": {"b": "bar"}}
+        table = self.lua.table_from(data, recursive=True)
+        self.assertEqual(table["a"]["a"], "foo")
+        self.assertEqual(table["b"]["b"], "bar")
+        self.lua.globals()["data"] = table
+        self.lua.eval("assert(data.a.a=='foo', 'failed')")
+        self.lua.eval("assert(data.b.b=='bar', 'failed')")
+        self.lua.eval("assert(type(data.a)=='table', 'failed, expect table, got '..type(data.a))")
+        self.lua.eval("assert(type(data.b)=='table', 'failed, expect table, got '..type(data.b))")
+        self.lua.execute("""function itertable(table) 
+            for k,v in pairs(table) do 
+                print(k)  
+                if type(v) == "table" then
+                    itertable(v)
+                else
+                    print(v)
+                end  
+            end 
+        end
+        print('\\n')
+        itertable(data)
+        """)
```

Comment:
The output of these is only printed into the test log and not validated anywhere. Maybe you could collect the output in a list and return it to Python for validation, instead of printing it?

Reply:
It was used for debugging. I'll fix it.